images/router/haproxy/reload-haproxy

```diff
@@ -12,8 +12,13 @@ if [ -f $pid_file ]; then
   old_pid=$(<$pid_file)
 fi
 
+# Assumes that the docker ip is the first entry in the hosts file.
+# This port is non-configurable via the router cmd, it is HAProxy specific.  It should
+# match the port used in the haproxy-config.template
+peer_name=$(awk 'NR==1 {print $1}' /etc/hosts):1937
```

Comment:
@ramr - yeah, the kicker is that the ids need to match across replicas because haproxy uses that to identify the servers during synchronization (so my static self label test failed).   The peer section labels need to be consistent among all peers with the -L flag identifying which peer you are. 

Unfortunately I'm back to the current solution since what I have to work with is the set of endpoints from the router service.

What I need is

```
peers mypeers
    peer haproxy1 192.168.0.1:1024
    peer haproxy2 192.168.0.2:1024
    peer haproxy3 10.2.0.1:1024
```

where each haproxy instance can be started with their own `-L <my label>` that matches one of those items in peers.

Suggestion box open.  Requirements 

1.  must be able to identify the label for the replica in the restart script for -L
1.  must be able to generate the peer section on all replicas based on the set of endpoints (with a label that matches item 1)

/cc @rajatchopra